src/main/java/redis/clients/jedis/CommandArguments.java

```diff
@@ -113,10 +122,25 @@ public CommandArguments key(Object key) {
     } else {
       throw new IllegalArgumentException("\"" + key.toString() + "\" is not a valid argument.");
     }
-    keys.add(key);
+
+    addKeyInKeys(key);
+
     return this;
   }
 
+  private void addKeyInKeys(Object key) {
```

Comment:
what are the sample cases we get the benefit from a collections.list??

Reply:
most Redis commands have single key. this way we are avoiding creation of an array just for an entry. so saving memory.